### PR TITLE
PR: Notify frontend of Matplotlib backend change

### PR DESCRIPTION
**Pull request by impact27**
Part of https://github.com/spyder-ide/spyder/pull/14199

### CODE REVIEW

**Comment by ccordoba12** [comment]
Hey @impact27, could you update this PR? It has merge conflicts now.

**Comment by ccordoba12** [comment]
Do you plan to open a new PR against master?

**Comment by impact27** [comment]
> Do you plan to open a new PR against master?

I am testing with 4.x for now but it should be easy to merge master when needed

**Comment by impact27** [comment]
Rebased against master

File: spyder_kernels/console/shell.py [hunk 1]
```diff
@@ -44,7 +47,18 @@ def _showtraceback(self, etype, evalue, stb):
             stb = ['']
         super(SpyderShell, self)._showtraceback(etype, evalue, stb)
 
-    # ---- For Pdb namespace integration
+    def enable_matplotlib(self, gui=None):
+        """Enable matplotlib."""
+        if gui is None or gui.lower() == "auto":
+            gui = automatic_backend()
+        gui, backend = super(SpyderShell, self).enable_matplotlib(gui)
```
Comment (ccordoba12) [inline spyder_kernels/console/shell.py:54]: Perhaps is better to use the methods we have in the kernel to set the Matplotlib backend, which handle errors gracefully? Or do you think that's not necessary in this case?

**Review by ccordoba12 — COMMENTED** [review]
Thanks @impact27 for your help with this!

**Comment by ccordoba12** [comment]
I'm also moving this to `master` because we're not planning to make more major changes in Spyder 5, as you know.

File: spyder_kernels/console/shell.py (hunk 1 above)
Comment (impact27) [reply spyder_kernels/console/shell.py:54]: This is the function currently called whenever the matplotlib backend is changed. I reimplement it to notify the backend. The methods we have in the kernel depends on this.

**Review by ccordoba12 — APPROVED** [review]
Thanks @impact27!

**Pull request merged**